solr/core/src/java/org/apache/solr/core/TransientSolrCoreCacheDefault.java

```diff
@@ -124,38 +142,29 @@ public SolrCore addCore(String name, SolrCore core) {
   
   @Override
   public Set<String> getLoadedCoreNames() {
-    return Collections.unmodifiableSet(transientCores.keySet());
+    return Collections.unmodifiableSet(transientCores.asMap().keySet());
   }
 
-  // Remove a core from the internal structures, presumably it 
-  // being closed. If the core is re-opened, it will be re-added by CoreContainer.
   @Override
   public SolrCore removeCore(String name) {
-    return transientCores.remove(name);
+    return transientCores.asMap().remove(name);
   }
 
-  // Get the core associated with the name. Return null if you don't want this core to be used.
   @Override
   public SolrCore getCore(String name) {
-    return transientCores.get(name);
+    return name == null ? null : transientCores.asMap().get(name);
```

Comment:
```suggestion
    return name == null ? null : transientCores.getIfPresent(name);
```

And thus the cache stats would actually be useful, thus remove or alter the javadocs you added above?

Reply:
Sounds good.